File: oracles/Java Oracles/02 - Minimum Value/Min.java
public class Min {

	public static int minimum(int[] values) {
		int min = values[0];
		for (int i = 1; i < values.length; i++)
			if (min > values[i])
				min = values[i];
		return min;
	}
	
	public static int min(int[] values) throws IllegalArgumentException {
		if (values.length == 0)
			throw new IllegalArgumentException();
		else
			return minimum(values);
	}
	
}

File: oracles/C Oracles/02 - Minimum Value/Min.c
#include "Min.h"

short int minimum(short int *values, unsigned char n) {
	short int min = values[0];
	unsigned char i;
	for (i = 1; i < n; i++)
		if (min > values[i])
			min = values[i];
	return min;
}

short int *min(short int *values, unsigned char n) {
	short int *min = (short int *) calloc(2, sizeof(short int));
	if (n > 0) {
		min[0] = 1;
		min[1] = minimum(values, n);
	} else {
		min[0] = 0;
	}
	return min;
}
